### ISSUE TITLE
EPIC: v2.0.0 release

### ISSUE THREAD

**Issue by smarshall-spitzbart** [cosmos/interchain-security#949]
# Problem

See description from https://github.com/cosmos/interchain-security/issues/933

Ie. we need to take all recent features, fixes, and refactors that have accumulated in main and issue a new release.

# Closing criteria

Cut a release of ICS with all recent accumulated squash commits in main. 

# Problem details

# Task list

```[tasklist]
### Must have
- [x] Merge https://github.com/cosmos/interchain-security/pull/931
- [x] Updated changelog in main, incl above
- [x] Hash out what migrations will be needed to upgrade gaia to this new version of ICS
- [x] Cut release
```

```[tasklist]
### Nice to have

```

**Comment by smarshall-spitzbart** [comment]
Per discussion with @mpoke this is now v2 release

**Comment by smarshall-spitzbart** [comment]
> Hash out what migrations will be needed to upgrade gaia to this new version of ICS

## v1.0.0 -> v2.0.0

To be complete, it makes sense to first analyze migrations required between v1.0.0 and v2.0.0 of ICS. The multiden versions aren't too far off the v1.0.0 linage. 

I've looked at [the diff between main and v1.0.0](https://github.com/cosmos/interchain-security/compare/v1.0.0...main), along with the PRs included in the changelog from main. From what I've found, if a consumer/provider were to upgrade from v1->v2, the only migrations required would be related to adding params. This is covered by the `Migratev1Tov2` methods for consumer/provider in #975.

TLDR: none of the PRs from [the v2.0.0 changelog entry](https://github.com/cosmos/interchain-security/blob/729fb54ece82474a8ce2a9e815939b6ece3d8a7c/CHANGELOG.md#prs-included-in-v200), should require state migrations besides those relevant to adding params. 

Note that https://github.com/cosmos/interchain-security/pull/963 makes it so that no key prefix related migrations are required. 

## Multiden versions (running in prod) -> v2.0.0

The migrations required for `v1.1.0-multiden` and `v1.2.0-multiden` to `v2.0.0` should simply be the migrations required for v1.0.0 -> v2.0.0, minus the migrations of individual squash commits that exist in each of those versions. I'll add funcs for these into #975

**Comment by smarshall-spitzbart** [comment]
Update: 

PRs that prevent the need for certain migrations:
- #963 <- prevents key prefix migration on consumers
- #980 <- prevents cons addr migration on provider

## Consumer migrations

With the above PRs merged, it actually seems like the consumer does not require migrations between `v1.2.0-multiden` and `v2.0.0`. Reasoning is that the former version includes both https://github.com/cosmos/interchain-security/commit/54e9852d3c89a2513cd0170a56c6eec894fc878d and https://github.com/cosmos/interchain-security/pull/833, which handle the addition of new consumer params properly.

## Provider migrations

#975 has been updated to reflect that the migration of `v1.1.0-multiden` to `v2.0.0` is the same as if migrating from `v1.0.0`. The migrations needed are:
- One new param added to provider from #931 
- Handling the fact that the consumer genesis state schema has changed, ie. we delete the only existing consumer genesis, Neutron
- Handling migrations from this fix, https://github.com/cosmos/interchain-security/pull/786

**Comment by smarshall-spitzbart** [comment]
^ it'll be useful to sanity check these conclusions, for now I will open up #975 for review

**Comment by smarshall-spitzbart** [comment]
RC is now cut https://github.com/cosmos/interchain-security/releases/tag/v2.0.0-rc1

**Comment by mpoke** [comment]
https://github.com/cosmos/interchain-security/releases/tag/v2.0.0 is cut.
